lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -375,6 +389,60 @@ function createPsuedoLocaleStrings(messages) {
   return psuedoLocalizedStrings;
 }
 
+/**
+ * @param {import('typescript').NamedDeclaration} node
+ * @return {string}
+ */
+function getIdentifier(node) {
+  if (!node.name || !tsc.isIdentifier(node.name)) throw new Error('no Identifier found');
+
+  return node.name.text;
+}
+
+/**
+ * @param {string} sourceStr String of the form 'const UIStrings = {...}'.
+ * @param {Record<string, string>} liveUIStrings The actual imported UIStrings object.
+ * @return {Record<string, ParsedUIString>}
+ */
+function parseUIStrings(sourceStr, liveUIStrings) {
+  const tsAst = tsc.createSourceFile('uistrings', sourceStr, tsc.ScriptTarget.ES2019, true, tsc.ScriptKind.JS);
+
+  const extractionError = new Error('UIStrings was not extracted correctly by the collect-strings regex.');
+  const uiStringsStatement = tsAst.statements[0];
+  if (tsAst.statements.length !== 1) throw extractionError;
+  if (!tsc.isVariableStatement(uiStringsStatement)) throw extractionError;
+
+  const uiStringsDeclaration = uiStringsStatement.declarationList.declarations[0];
+  if (!tsc.isVariableDeclaration(uiStringsDeclaration)) throw extractionError;
+  if (getIdentifier(uiStringsDeclaration) !== 'UIStrings') throw extractionError;
+
+  const uiStringsObject = uiStringsDeclaration.initializer;
+  if (!uiStringsObject || !tsc.isObjectLiteralExpression(uiStringsObject)) throw extractionError;
+
+  /** @type {Record<string, ParsedUIString>} */
+  const parsedMessages = {};
+
+  for (const property of uiStringsObject.properties) {
+    const key = getIdentifier(property);
+
+    // Use live message to avoid having to e.g. concat strings broken into parts.
+    const message = liveUIStrings[key];
+
+    // @ts-ignore - Not part of the public tsc interface yet.
+    const jsDocComments = tsc.getJSDocCommentsAndTags(property);
+    if (jsDocComments.length === 0) throw new Error(`No Description for message "${message}"`);
```

Comment:
can we get rid of the message inside `computeDescription` then and make this the "Missing description comment" one?

Reply:
> can we get rid of the message inside `computeDescription` then and make this the "Missing description comment" one?

hmm, do you mean like check that either `ast.comment` or `ast.tags` is defined here? Or output an empty `description` from `computeDescription` and throw here if it's the empty string?

Reply:
yeah I meant this is a duplicate case and error message of the fallback in `computeDescription`, could `computeDescription` accept `ast|undefined` and do the check there?

Reply:
> could `computeDescription` accept `ast|undefined` and do the check there?

got it